components/gitpod-protocol/src/util/gitpod-host-url.ts

```diff
@@ -24,21 +23,11 @@ const workspaceUrlPrefixRegex = RegExp(`^(([0-9]{4,6}|debug)-)?${baseWorkspaceID
 export class GitpodHostUrl {
     readonly url: URL;
 
-    constructor(urlParam?: string | URL) {
-        if (urlParam === undefined || typeof urlParam === "string") {
-            this.url = new URL(urlParam || "https://gitpod.io");
-            this.url.search = "";
-            this.url.hash = "";
-            this.url.pathname = "";
-        } else if (urlParam instanceof URL) {
-            this.url = urlParam;
-        } else {
-            log.error("Unexpected urlParam", { urlParam });
-        }
-    }
-
-    static fromWorkspaceUrl(url: string) {
-        return new GitpodHostUrl(new URL(url));
+    constructor(urlParam?: string) {
+        this.url = new URL(urlParam || "https://gitpod.io");
```

Comment:
@geropl this one does not make sense in dedicated? Should we rather make arg mandatory?

Reply:
only in tests, i remove a fallback